nextjs-app/app/components/PageBuilder.tsx

```diff
@@ -1,48 +1,107 @@
+"use client";
+
+import { createDataAttribute, SanityDocument } from "next-sanity";
+import { useOptimistic } from "next-sanity/hooks";
 import Link from "next/link";
 
 import BlockRenderer from "@/app/components/BlockRenderer";
 import { Page } from "@/sanity.types";
+import { createDataAttributeConfig } from "@/sanity/lib/utils";
 import { studioUrl } from "@/sanity/lib/api";
 
 type PageBuilderPageProps = {
   page: Page;
 };
 
+type PageBuilderSection = {
+  _key: string;
+  _type: string;
+};
+
+type PageData = {
+  _id: string;
+  _type: string;
+  pageBuilder?: PageBuilderSection[];
+};
+
 /**
  * The PageBuilder component is used to render the blocks from the `pageBuilder` field in the Page type in your Sanity Studio.
  */
-export default function PageBuilder({ page }: PageBuilderPageProps) {
-  if (page?.pageBuilder && page.pageBuilder.length > 0) {
-    return (
-      <>
-        {page.pageBuilder.map((block: any, index: number) => (
-          <BlockRenderer key={block._key} index={index} block={block} />
-        ))}
-      </>
-    );
-  }
-
-  // If there are no blocks in the page builder.
+
+function renderSections(pageBuilderSections: PageBuilderSection[], page: Page) {
+  return (
+    <div
+      data-sanity={createDataAttribute(
+        createDataAttributeConfig({
+          id: page._id,
+          type: page._type,
+          path: `pageBuilder`,
+        })
+      ).toString()}
```

Comment:
This could be something like this using the method I described elsewhere
```suggestion
      data-sanity={dataAttr({
        id: page._id,
        type: page._type,
        path: `pageBuilder`,
      }).toString()}
```